modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/deployer/ObjectDefinitionDeployerImpl.java

```diff
@@ -328,13 +362,50 @@ private void _initCustomObjectDefinition(
 				restContextPath,
 				_bundleContext.registerService(
 					Application.class,
-					new ObjectEntryApplication(_objectEntryOpenAPIResource),
+					new ObjectEntryApplication(
+						_objectEntryOpenAPIResourceProvider),
 					properties));
 		}
 		else {
 			applicationServiceRegistration.setProperties(properties);
 		}
 
+		_scopedServiceRegistrationsMap.compute(
+			restContextPath,
+			(key1, serviceRegistrationsMap) -> {
+				if (serviceRegistrationsMap == null) {
+					serviceRegistrationsMap = new HashMap<>();
+				}
+
+				serviceRegistrationsMap.computeIfAbsent(
+					objectDefinition.getCompanyId(),
+					key2 -> Arrays.asList(
+						_bundleContext.registerService(
+							ObjectEntryOpenAPIResource.class,
+							new ObjectEntryOpenAPIResourceImpl(
+								_bundleContext, _dtoConverterRegistry,
+								_objectActionLocalService, objectDefinition,
+								_objectDefinitionLocalService,
+								_objectFieldLocalService,
+								_objectRelationshipLocalService,
+								_openAPIResource,
+								_systemObjectDefinitionMetadataRegistry),
+							HashMapDictionaryBuilder.<String, Object>put(
+								"companyId", objectDefinition.getCompanyId()
+							).put(
+								"openapi.resource", "true"
+							).put(
+								"osgi.jaxrs.application.select",
+								"(osgi.jaxrs.name=" + osgiJaxRsName + ")"
+							).put(
+								"osgi.jaxrs.name",
+								objectDefinition.getOSGiJaxRsName(
+									"ObjectEntryOpenAPIResource")
```

Comment:
I don't get these properties. Maybe I am missing something (today is not my best day understanding code, to be honest 😅😅)

What I don't get is that, apart from the name of the class `ObjectEntryOpenAPIResourceImpl`, the class does not represent a JSXRS resource, right? I think most of the properties are not necessary. Maybe the unique important property here is `osgi.jaxrs.name` as it is the property used to filter in the serviceTrackerMap of the class `ObjectEntryOpenAPIResourceProviderImpl`

Am I correct?

Anyway, this is another silly comment as the properties will not affect the functionality. It is only information attached to the component 😁

Reply:
Yes, you are totally right. In fact, I am going to rename the property to avoid confusions about the osgi.jarxs.name as it is not necessary, it is only necessary for the Provider.

The point of adding those properties was in the past by managing the name of the applications based on the JAX-RS properties, but finally I adapted it to the same functionality as it was being done based on the JaxRS runtime and paths